distributed/scheduler.py

```diff
@@ -1430,16 +1434,24 @@ def _transition(self, key, finish: str, stimulus_id: str, *args, **kwargs):
         Scheduler.transitions : transitive version of this function
         """
         try:
-            recommendations = {}  # type: ignore
-            worker_msgs = {}  # type: ignore
-            client_msgs = {}  # type: ignore
-
             ts: TaskState = self.tasks.get(key)  # type: ignore
             if ts is None:
-                return recommendations, client_msgs, worker_msgs
+                return {}, {}, {}
             start = ts._state
             if start == finish:
-                return recommendations, client_msgs, worker_msgs
+                return {}, {}, {}
+
+            # Notes:
+            # - in case of transition through released, this counter is incremented by 2
+            # - this increase happens before the actual transitions, so that it can
+            #   catch potential infinite recursions
+            self.transition_counter += 1
+            if self.validate and self.transition_counter_max:
+                assert self.transition_counter < self.transition_counter_max
```

Comment:
Might be nice if this raised something like a `TransitionCounterMaxExceeded` error, to be consistent with workers

Reply:
yeah, but scheduler doesn't have anything like InvalidTransitionError on the worker